cli_tools/gce_vm_image_import/importer/processor.go

```diff
@@ -37,24 +42,62 @@ type processorProvider interface {
 
 type defaultProcessorProvider struct {
 	ImportArguments
-	imageClient   createImageClient
+	computeClient daisyCompute.Client
 	diskInspector disk.Inspector
 }
 
 func (d defaultProcessorProvider) provide(pd persistentDisk) (processor, error) {
 	if d.DataDisk {
-		return newDataDiskProcessor(pd, d.imageClient, d.Project,
+		return newDataDiskProcessor(pd, d.computeClient, d.Project,
 			d.Labels, d.StorageLocation, d.Description,
 			d.Family, d.ImageName), nil
 	}
+	var err error
+	pd, err = d.inspectDisk(pd)
+	if err != nil {
+		return nil, err
+	}
+	return newBootableDiskProcessor(d.ImportArguments, pd)
+}
+
+func (d defaultProcessorProvider) inspectDisk(pd persistentDisk) (persistentDisk, error) {
+	var inspectionResult disk.InspectionResult
+	var err error
 	if d.Inspect && d.diskInspector != nil {
 		log.Printf("Running experimental disk inspections on %v.", pd.uri)
-		inspectionResult, err := d.diskInspector.Inspect(pd.uri)
+		inspectionResult, err = d.diskInspector.Inspect(pd.uri)
 		if err != nil {
-			log.Printf("Inspection error=%v", err)
-		} else {
-			log.Printf("Inspection result=%v", inspectionResult)
+			log.Printf("Disk inspection error=%v", err)
+			return pd, daisy.Errf("Disk inspection error: %v", err)
 		}
+
+		log.Printf("Disk inspection result=%v", inspectionResult)
 	}
-	return newBootableDiskProcessor(d.ImportArguments, pd)
+
+	// If UEFI_COMPATIBLE is enforced in user input args (by d.ImportArguments.UefiCompatible),
+	// then it has been honored in inflation stage, so no need to create a new disk here.
+	// Only create new disk with UEFI_COMPATIBLE when inspection result tells us to do it.
+	if !d.ImportArguments.UefiCompatible && inspectionResult.HasEFIPartition {
+		diskName := fmt.Sprintf("disk-%v-uefi", d.ImportArguments.ExecutionID)
+		err := d.computeClient.CreateDisk(d.ImportArguments.Project, d.ImportArguments.Zone, &compute.Disk{
+			Name:            diskName,
+			SourceDisk:      pd.uri,
+			GuestOsFeatures: []*compute.GuestOsFeature{{Type: "UEFI_COMPATIBLE"}},
+		})
```

Comment:
Creating a new disk in order to set GuestOSFeatures doesn't belog to inspector. Inspector should just return info on the disk/image, it should not make updates. Updates belong to the processor or a separate step.

Reply:
Sounds great, I moved it into bootableDiskProcessor.process()